internal/ctrlutils/ingress-status.go

```diff
@@ -347,33 +357,46 @@ func UpdateTCPIngress(ctx context.Context, logger logr.Logger, svc file.FService
 		log.Debugf("Updating TCP ingress route name %s namespace %s", name, namespace)
 
 		ingCli := kiccli.ConfigurationV1beta1().TCPIngresses(namespace)
-		curIng, err := ingCli.Get(ctx, name, metav1.GetOptions{})
-		if err != nil || curIng == nil {
-			if errors.IsNotFound(err) {
-				return nil
-			}
 
-			return fmt.Errorf("failed to fetch TCP Ingress %v/%v: %w", namespace, name, err)
-		}
+		retry := 0
+		for retry < statusUpdateRetry {
+			curIng, err := ingCli.Get(ctx, name, metav1.GetOptions{})
+			if err != nil || curIng == nil {
+				if errors.IsNotFound(err) {
+					return nil
+				}
 
-		var status []apiv1.LoadBalancerIngress
-		sort.SliceStable(status, lessLoadBalancerIngress(status))
-		curIPs := curIng.Status.LoadBalancer.Ingress
+				log.Errorf("failed to fetch TCPIngress %v/%v: %v", namespace, name, err)
+				time.Sleep(statusUpdateWaitTick)
+				retry++
+				continue
+			}
 
-		status = SliceToStatus(ips)
-		if ingressSliceEqual(status, curIPs) {
-			log.Debugf("no change in status, update tcp ingress skipped")
-			return nil
-		}
+			var status []apiv1.LoadBalancerIngress
+			sort.SliceStable(status, lessLoadBalancerIngress(status))
```

Comment:
```suggestion
			
```
status is an empty object, does not need SliceStable.

Reply:
good catch :+1: I'm kinda surprised linting didn't catch that?

resolved by 2b14fa2562f8c18b0581bb7ab897fe69491960c3